lightning/src/ln/onchaintx.rs

```diff
@@ -0,0 +1,696 @@
+			let input_material = Readable::read(reader)?;
+			per_input_material.insert(outpoint, input_material);
+		}
+		Ok(Self { height_timer, feerate_previous, soonest_timelock, per_input_material })
+	}
+}
+
+#[derive(PartialEq)]
+pub(super) enum InputDescriptors {
+	RevokedOfferedHTLC,
+	RevokedReceivedHTLC,
+	OfferedHTLC,
+	ReceivedHTLC,
+	RevokedOutput, // either a revoked to_local output on commitment tx, a revoked HTLC-Timeout output or a revoked HTLC-Success output
+}
+
+macro_rules! subtract_high_prio_fee {
+	($self: ident, $fee_estimator: expr, $value: expr, $predicted_weight: expr, $used_feerate: expr) => {
+		{
+			$used_feerate = $fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::HighPriority);
+			let mut fee = $used_feerate * ($predicted_weight as u64) / 1000;
+			if $value <= fee {
+				$used_feerate = $fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Normal);
+				fee = $used_feerate * ($predicted_weight as u64) / 1000;
+				if $value <= fee {
+					$used_feerate = $fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Background);
+					fee = $used_feerate * ($predicted_weight as u64) / 1000;
+					if $value <= fee {
+						log_error!($self, "Failed to generate an on-chain punishment tx as even low priority fee ({} sat) was more than the entire claim balance ({} sat)",
+							fee, $value);
+						false
+					} else {
+						log_warn!($self, "Used low priority fee for on-chain punishment tx as high priority fee was more than the entire claim balance ({} sat)",
+							$value);
+						$value -= fee;
+						true
+					}
+				} else {
+					log_warn!($self, "Used medium priority fee for on-chain punishment tx as high priority fee was more than the entire claim balance ({} sat)",
+						$value);
+					$value -= fee;
+					true
+				}
+			} else {
+				$value -= fee;
+				true
+			}
+		}
+	}
+}
+
+
+/// OnchainTxHandler receives claiming requests, aggregates them if it's sound, broadcast and
+/// do RBF bumping if possible.
+#[derive(Clone)]
+pub struct OnchainTxHandler {
+	destination_script: Script,
+
+	// Used to track claiming requests. If claim tx doesn't confirm before height timer expiration we need to bump
+	// it (RBF or CPFP). If an input has been part of an aggregate tx at first claim try, we need to keep it within
+	// another bumped aggregate tx to comply with RBF rules. We may have multiple claiming txn in the flight for the
+	// same set of outpoints. One of the outpoints may be spent by a transaction not issued by us. That's why at
+	// block connection we scan all inputs and if any of them is among a set of a claiming request we test for set
+	// equality between spending transaction and claim request. If true, it means transaction was one our claiming one
+	// after a security delay of 6 blocks we remove pending claim request. If false, it means transaction wasn't and
+	// we need to regenerate new claim request with reduced set of still-claimable outpoints.
```

Comment:
If you're gonna fix a spelling mistake in a mostly-move, can you do it in a separate commit so that --color-moved doesn't barf?

Reply:
I don't remember to do a spelling fix there ? What git options did you use there?

Reply:
git show --color-moved highlights this line as a different color.